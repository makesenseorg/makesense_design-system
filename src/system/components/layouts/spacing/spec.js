import { shallowMount } from "@vue/test-utils";
import Comp from "./Spacer.vue";

describe("Spacer.vue", () => {
  it("matches snapshot", () => {
    const wrapper = shallowMount(Comp);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("displays slots", () => {
    const wrapper = shallowMount(Comp, {
      slots: {
        default: '<p>a</p><p>b</p>',
      },
    });

    expect(wrapper.text()).toContain("ab");
  });

  it("displays elements inline", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        inline: true,
      },
    });

    expect(wrapper.classes()).toContain("spacer--inline");
  });
});
