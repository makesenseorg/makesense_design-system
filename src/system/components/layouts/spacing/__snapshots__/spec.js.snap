// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer.vue matches snapshot 1`] = `
<div
  class="spacer spacer--spacing-m spacer--align-start spacer--distribute-start spacer--stack"
/>
`;
